drivers/video/video_mcux_mipi_csi2rx.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mipi_csi2rx
+
+#include <fsl_mipi_csi2rx.h>
+
+#include <zephyr/drivers/video.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mipi_csi);
+
+/*
+ * Two data lanes are set by default as 2-lanes camera sensors are
+ * more common and more performant but single lane is also supported.
+ */
+#define DEFAULT_MIPI_CSI_NUM_LANES 2
+#define DEFAULT_CAMERA_FRAME_RATE  30
```

Comment:
Is there a value to making this a Kconfig. Write now the user has to edit this driver file to change the default frame rate.